server/src/main/java/org/elasticsearch/search/crossproject/IndexExpressionsRewriter.java

```diff
@@ -127,7 +128,7 @@ private static IndexRewriteResult rewriteUnqualifiedExpression(
         Set<String> allProjectAliases
     ) {
         String localExpression = null;
-        final List<String> rewrittenExpressions = new ArrayList<>();
+        final Set<String> rewrittenExpressions = new LinkedHashSet<>();
```

Comment:
It's fine for these to be Sets instead of Lists and it closer matches the ResolvedIndexExpressions class.